### PR TITLE
feat: versions 8.4/3.12.0, reviewable steps

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

The more reviewable part of #3151.

This PR includes some smaller changes for cutting versions 8.4/3.12.0, so that they are more easily reviewed.

Once this is merged, #3183 can be reviewed/merged.

There will likely be broken redirects on this one, until I can sniff them all out.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: spec/regression/versionMapping.spec.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ test("optimize docs cross-link to main docs", async ({ page }) => {
     "/optimize/apis-tools/optimize-api/optimize-api-authentication/"
   );
 
-  await expect(page).toHaveTitle(/Authorization \| Camunda 8 Docs/);
+  await expect(page).toHaveTitle(/Authentication \| Camunda 8 Docs/);
```
Comment (akeller) [inline spec/regression/versionMapping.spec.ts:23]: Good spot! That title did change!

**Review by akeller — APPROVED** [review]
As I was reviewing this, I realized this release is special in that no versions will be moved to unsupported status. 

No relevant to this PR, but I needed to get it out of my brain.

**Comment by akeller** [comment]
> There will likely be broken redirects on this one, until I can sniff them all out.

Redirect issues look all isolated to APIs & tools section and reducing the troubleshooting/common problems pages in Identity down to one. LMK if I can help.

File: spec/regression/versionMapping.spec.ts (hunk 1 above)
Comment (pepopowitz) [reply spec/regression/versionMapping.spec.ts:23]: You have no idea how long I stared at that failing test trying to pick out the change 😵‍💫

**Comment by pepopowitz** [comment]
> > There will likely be broken redirects on this one, until I can sniff them all out.
> 
> Redirect issues look all isolated to APIs & tools section and reducing the troubleshooting/common problems pages in Identity down to one. LMK if I can help.

These specifically are resolved! Now I'm on to the _actual_ redirect issues -- https://github.com/camunda/camunda-docs/actions/runs/7453148520/job/20277921566

**Pull request merged**